src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistoryArray = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        stateHistoryArray.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        stateHistoryArray.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        stateHistoryArray.push({ ...stateCopy });
+        break;
+
+      default:
+        break;
```

Comment:
[CODE STYLE]: The default case in a switch statement is typically used for error handling. It's good practice to include a message or throw an error to handle unexpected cases.

Reply:
Fix this